core/model/customlist.py

```diff
@@ -317,8 +326,8 @@ def entries_for_work(self, work_or_edition):
         )
         return qu
 
-    def update_size(self):
-        self.size = len(self.entries)
+    def update_size(self, db: Session):
+        self.size = CustomList.list_entries(db, self.id).count()
```

Comment:
Will this cause an already run query to be re-run unnecessarily?

Reply:
I don't think so, but I'm not 100% sure.   I'm not sure if the previous approach was causing a query to run behind the scenes ie 
```
self.size = len(self.entries)
```
(I'm not familiar with the internal workings of sqlalchemy). 

That said, this method code is only called once in the entire codebase (tests excluded) and it is called in a cron job script it should have no impact whatsoever on the user experience.